protocol/x/affiliates/keeper/keeper.go

```diff
@@ -41,3 +49,154 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 }
 
 func (k Keeper) InitializeForGenesis(ctx sdk.Context) {}
+
+func (k Keeper) RegisterAffiliate(
+	ctx sdk.Context,
+	referee string,
```

Comment:
Unrelated to this PR: could you create a linear ticket to add a e2e test case that `Alice` cannot sign a transaction on behalf of Bob (that registers `Bob` as refererred by `Carl`)

Reply:
[Done](https://linear.app/dydx/issue/OTE-730/add-e2e-tests-for-affiliates)